### ISSUE TITLE
Deployment problems for the `docker-compose` file following the official guide

### ISSUE THREAD

**Issue by AleixMT** [ComPlat/chemotion_ELN#1331]
I am setting up a production environment.

# Failure Information 

[I am following the instructions depicted in the official documentation to deploy Chemotion as a Docker image](https://chemotion.net/docs/eln/install_configure). I have cloned the repository and have docker CLI installed on my machine. I have run the command `docker-compose -f docker-compose.dev.yml up postgres app webpacker` but it gives me an error about the `docker-compose.dev.yml` file having invalid configuration.

## Context

The context is trying to change [this unofficial docker image](https://hub.docker.com/r/ptrxyz/chemotion) with the official ones that you provide since the antivirus have found "suspicious" files in the overlays of this image. Apparently there is an overlay of that image with Wine and some `.exe` files are making the antivirus complain. It could be that this unofficial image comes with malware. I have a few questions regarding this:
- Do any of the components of Chemotion execute Wine in any way? 
- Do anybody knows if [this user](https://github.com/ptrxyz) can be trusted? [He did not provide the docker files requested](https://github.com/ptrxyz/chemotion/issues/25). 

If you could provide a little bit of context regarding this I would be very grateful. 

I have now stopped the containers with suspicious files and even if the unofficial image can be trusted I would prefer to use your official `docker-compose.yml` file. Do you know why `docker` is complaining?

I am using Docker
 
- OS: 20.04.5 LTS (Focal Fossa) 
- Docker version: 20.10.21, build 20.10.21-0ubuntu1~20.04.2
- ELN version, or branch, or commit: `main` branch, commit ID: `f33ce10a619ddaef67bff7f08dadd4b852b34e98`

## Steps to Reproduce

1. step 1: Install Ubuntu
2. step 2: Install Docker and git: `sudo apt install -y docker git`
3. step 3: Clone this repository: `git clone https://github.com/ComPlat/chemotion_ELN`
4. step 4: Navigate to the repo `cd chemotion_ELN`
5. step 5: Issue conflicting being with privileges: `docker-compose -f docker-compose.dev.yml up postgres app webpacker`

## Failure Logs / Screenshots

![image](https://github.com/ComPlat/chemotion_ELN/assets/23342150/11721e84-7304-472a-abea-0cd5aff3b103)

**Comment by AleixMT** [comment]
Thank you for helping and maintaing Chemotion!

**Comment by JanCBrammer** [comment]
@AleixMT, thanks for the issue. @ptrxyz, can you have a look?

**Comment by AleixMT** [comment]
@ptrxyz Can you provide the dockerfiles and/or an explanation of what could be happening?

**Comment by baolanlequang** [comment]
@AleixMT I think @ptrxyz will help you with more details. But I hope this information will hope you. 

In the `docker-compose.yml` file, there is one service called `msconvert`. This service is from [here](https://hub.docker.com/r/chambm/pwiz-skyline-i-agree-to-the-vendor-licenses), it is used for converting Mass Spectrometry data and it needs Windows environment files in itself.

If you don't want to use the function that displays Mass Spectrometry, you can remove this service from `docker-compose.yml` file.

**Comment by ptrxyz** [comment]
Hello @AleixMT,

I personally would not trust anyone on the web, least of all myself. Not sure what your virus scanner detects, the software we use is the one mentioned by @baolanlequang and it requires the Wine environment.

Beside the application itself, the docker images I provide contain a lot of glue code that handles synchronization between the services, does some configuration (i.e. merging env variables to the right config files, and sets defaults to match the compose file) and some optimizations to keep the images small (layers are squashed and we try to built services from a common shared base that might be bigger for a single service, but smaller when all services are used and the layers can be shared). So I am afraid the Dockerfiles on their own would not even help you a lot without the glue code mentioned. Feel free to inspect the images though, there is nothing secret in them: `/embed/run.sh` is the entry point launching a batch of boot-up scripts in `/embed/scripts` depending on the role of the container.

**Comment by PiTrem** [comment]
Hi @AleixMT 

you can check there how the images are build: https://github.com/ptrxyz/chemotion-build 

Also, if you wish I can add you to our mailing list of __Chemotion ELN__ release announcement where images are also listed.

**Comment by AleixMT** [comment]
So what I finally did to solve my problem was to deploy chemotion using the same docker images that @ptrxyz provided but I deleted from the `docker-compose.yml` the `msconvert` module. 

Probably the antivirus was giving a false positive since it was detecting files from Wine, but I just wanted to make the antivirus stop complaining. 

Chemotion keeps working perfectly even if it is deployed without this module. 

Part of the code that I use to deploy my chemotion instance can be found in this [repository](https://github.com/ICIQ-DMP/ChemotionDeploy)
